src/uucore/src/lib/mods/update_control.rs

```diff
@@ -0,0 +1,136 @@
+//! Implement GNU-style update functionality.
+//!
+//! - pre-defined [`clap`-Arguments][1] for inclusion in utilities that
+//!   implement updates
+//! - determination of the [update mode][2]
+//!
+//! Update-functionality is implemented by the following utilities:
+//!
+//! - `cp`
+//! - `mv`
+//!
+//!
+//! [1]: arguments
+//! [2]: `determine_update_mode()`
+//!
+//!
+//! # Usage example
+//!
+//! ```
+//! #[macro_use]
+//! extern crate uucore;
+//!
+//! use clap::{Command, Arg, ArgMatches};
+//! use uucore::update_control::{self, UpdateMode};
+//!
+//! fn main() {
+//!     let matches = Command::new("command")
+//!         .arg(update_control::arguments::update())
+//!         .arg(update_control::arguments::update_no_args())
+//!         .get_matches_from(vec![
+//!             "command", "--update=older"
+//!         ]);
+//!
+//!     let update_mode = update_control::determine_update_mode(&matches);
+//!
+//!     // handle cases
+//!     if update_mode == UpdateMode::ReplaceIfOlder {
+//!         // do
+//!     } else {
+//!         unreachable!()
+//!     }
+//! }
+//! ```
+use clap::ArgMatches;
+
+// Available update mode
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum UpdateMode {
+    // --update=`all`, ``
+    ReplaceAll,
+    // --update=`none`
+    ReplaceNone,
+    // --update=`older`
+    // -u
+    ReplaceIfOlder,
+}
+
+pub mod arguments {
+    use clap::ArgAction;
+
+    pub static OPT_UPDATE: &str = "update";
+    pub static OPT_UPDATE_NO_ARG: &str = "u";
+
+    // `--update` argument, defaults to `older` if no values after provided
+    pub fn update() -> clap::Arg {
+        clap::Arg::new(OPT_UPDATE)
+            .long("update")
+            .help("move only when the SOURCE file is newer than the destination file or when the destination file is missing")
+            .value_parser(["", "none", "all", "older"])
```

Comment:
What's the purpose of the empty string? GNU cp/mv don't allow to use `--update=` and show an error. So my suggestion is to remove it:
```suggestion
            .value_parser(["none", "all", "older"])
```